firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java

```diff
@@ -43,7 +43,9 @@
 
     this.storageRef = storageRef;
     this.pendingResult = pendingResult;
-
+    if (storageRef.getRoot().getName().equals(storageRef.getName())) {
+      throw new IllegalArgumentException("Cannot get download url at the root of a bucket.");
```

Comment:
Can you change this to use the method name (getDownloadUrl())? Furthermore, does the same problem exists if a user calls getMetadata()?

Reply:
Can you make the error message something like: "getDownloadUrl() is not supported at the root of the Storage bucket."

Reply:
Done